design-proposals/klip-42-schema-migrations-tool.md

```diff
@@ -168,9 +170,12 @@ Current version of schema: 1
 Migrating schema to version 2 - Add users
 ```
 
-All migration files will support only integer versions (no decimal versions). Integer versions are easier to sort when are found in the file name. Also, there are
+All migration files will support only integer versions (no decimal versions). Integer versions are easier to sort when found in the file name. Also, there are
 too few cases that require decimal versioning (i.e. `1.1`) in schema changes. We don't expect users to use decimal versions on ksqlDB migrations.
 
+Version numbers need not be consecutive. In the event that earlier version numbers are added after a later version has already been migrated
+to, the tool will ignore the newly added versions by default, though we will support a flag to change this behavior and enable out-of-order migrations.
```

Comment:
Flyway allows this: https://flywaydb.org/documentation/configuration/parameters/outOfOrder

If we think it's confusing to allow out of order migrations, we can require that version numbers be consecutive. If we do this, then we should remove the `-v` option on `initialize` since the initial state must always have version 1 (or 0? Is 0 is valid version number?)

Reply:
Oh, I didn't read that part. Then let's do it :)